src/components/LocationPage/ShareButtons.js

```diff
@@ -0,0 +1,79 @@
+import React, { Fragment, useState, useRef } from 'react';
+import useOutsideClickHandler from '../../common/utils/useOutsideClickHandler';
+import SocialButtons from './SocialButtons';
+import {
+  SaveOrShareContainer,
+  SaveOrShareButton,
+  DesktopButtonsWrapper,
+  MobileButtonsWrapper,
+} from './ShareButtons.style';
+import useMediaQuery from '@material-ui/core/useMediaQuery';
+import { useTheme } from '@material-ui/core/styles';
+
+const InnerContent = props => {
+  const { iconSize, setShowShareIcons, showShareIcons } = props;
+
+  return (
+    <Fragment>
+      <SaveOrShareContainer>
+        <SaveOrShareButton
+          onClick={() => {
+            setShowShareIcons(false);
+          }}
+        >
+          Save
+        </SaveOrShareButton>
+        <SaveOrShareButton
+          isLast
+          onClick={() => {
+            setShowShareIcons(!showShareIcons);
+          }}
+        >
+          Share
+        </SaveOrShareButton>
+      </SaveOrShareContainer>
+      {showShareIcons && <SocialButtons iconSize={iconSize} />}
+    </Fragment>
+  );
+};
+
+const ShareButtons = props => {
+  const { isFirst, chartAbove } = props;
+  const [showShareIcons, setShowShareIcons] = useState(false);
+
+  const shareButtonsRef = useRef(null);
+
+  useOutsideClickHandler(shareButtonsRef, setShowShareIcons);
```

Comment:
If we can switch to https://material-ui.com/components/click-away-listener/ I suspect we can get rid of `shareButtonsRef` and move `showShareIcons` into the `InnerContent` component?

Reply:
implemented clickAwayListener and got rid of shareButtonsRef! I kept showShareIcons because it is used by clickAwayListener